doc/guide/api-cockpit.xml

```diff
@@ -1862,6 +1862,366 @@ handle = file.watch(callback);
+
+      <para>All rows with callbacks will be invoked to regenerate all the data. The
+        <link linkend="cockpit-grid-onnotify"><code>grid.onnotify</code></link> event will be triggered.
+        It is not necessary to call this function after a call of the
+        <link linkend="cockpit-grid-move"><code>grid.move()</code></link> method.</para>
+
+    </refsection>
+
+    <refsection id="cockpit-grid-move">
+      <title>grid.move()</title>
+<programlisting>
+  grid.move(beg[, end])
+</programlisting>
+
+      <para>Move the grid to new <code>beg</code> and <code>end</code> range. Data will be
+        discarded from the rows and <link linkend="cockpit-grid-sync"><code>grid.sync()</code></link>
+        will be called to load or reload series data for the new range of offsets.</para>
+
+      <para>If <code>end</code> is not specified it will be set to <code>beg</code>. If <code>beg</code>
+        and/or <code>end</code> are negative (including negative zero) then they will be set to the
+        number of intervals prior to the current time taken as an interval.</para>
+
+      <para>If the grid is <link linkend="cockpit-grid-walk">walking</link> then calling this function
+        will cause it to stop walking.</para>
+    </refsection>
+
+    <refsection id="cockpit-grid-walk">
+      <title>grid.walk()</title>
+<programlisting>
+  grid.walk()
+</programlisting>
+
+      <para>Move the grid forward every
+        <link linkend="cockpit-grid-interval"><code>grid.interval</code></link> milliseconds. To stop
+        moving forward, call <link linkend="cockpit-grid-move"><code>grid.move()</code></link>.</para>
+    </refsection>
+
+    <refsection id="cockpit-grid-notify">
+      <title>grid.notify()</title>
+<programlisting>
+  grid.notify(index, count)
+</programlisting>
+
+      <para>This function is called to have rows with callbacks recalculate their data. It is not
+        normally necessary to call this function, as it will be invoked automatically when new
+        series data is available or has been loaded. This function triggers the
+        <link linkend="cockpit-grid-onnotify"><code>grid.onnotify</code></link> event.</para>
+
+    </refsection>
+
+    <refsection id="cockpit-grid-onnotify">
+      <title>grid.onnotify</title>
+<programlisting>
+  $(grid).on("notify", function(index, count) { ... });
+</programlisting>
+
+      <para>An event that is triggered when same part of the series data in grid changes. The
+        <code>index</code> is the row index where things changed, and the <code>count</code>
+        is the length of the data that changed.</para>
+
+    </refsection>
+
+    <refsection id="cockpit-grid-close">
+      <title>grid.close()</title>
+<programlisting>
+  grid.close()
+</programlisting>
+
+      <para>Close the grid, and stop updating the rows.</para>
+    </refsection>
+
+    <refsection id="cockpit-grid-rows">
+      <title>grid.rows</title>
```

Comment:
I think we might be better off not exposing grid.rows as part of the API.  Clients can easily build their own collections of rows, and that is actually cleaner, I think.  See my comment in the cpu monitor code.